### ISSUE TITLE
Passing data from numpy trough the intermediate API and getting back the cross entropy

### ISSUE THREAD

**Issue by botev** [apache/incubator-mxnet#5189]
There are quite a few questions which are unclear of how to do. First, I want to initalize the model without any data given/binded. This seems quite difficult and could not find an example.

Secondly, I want to be able to pass numpy data to the forward and backward methods of the intermediate API, rather than using the `fit` method, or alternatively is there anyway to get a sensible callback which to return the cross entorpy + the time taken for the iteration?

My code currently looks like this:
```
    x = mx.symbol.Variable("inputs")
    y = mx.symbol.Variable("targets")

    h = x
    tanh = net.nonlinearity_from_string("tanh")
    softmax = net.nonlinearity_from_string("softmax")
    for i in range(1, len(arch_specs)-1):
        h = mx.symbol.FullyConnected(h, name="ff_" + str(i), num_hidden=arch_specs[i])
        h = tanh(h)
    h = mx.symbol.FullyConnected(h, name="ff_" + str(len(arch_specs)-1), num_hidden=arch_specs[-1])
    h = softmax(h, y)

    mod = mx.mod.Module(h, data_names=["inputs"], label_names=["targets"])
    mod.init_optimizer(optimizer_params={'learning_rate': learning_rate})
    metric = mx.metric.create('ce')

    for i in range(iterations):
        i1 = (i * batch_size) % num_samples
        # print(i1, i2)
        start_time = time.time()
        mod.forward(x[i1:i1 + batch_size])
        mod.update_metric(metric, y[i1:i1+batch_size])
        mod.backward()
        mod.update()
        train_time[i] = time.time() - start_time
        loss[i] = metric.get_name_value()["acc"]
        if debug:
            print("Train [{}]: {:.2f} - {:.3e}".format(i, train_time[i], loss[i]))
```

I'm not totally sure if this is the correct way, however atm on the line for creating the optimizer I get:
```
File "/home/alex/work/python/deep-bench/deep_bench/frameworks/mxnet/ff.py", line 19, in build_ff_net
    mod.init_optimizer(optimizer_params={'learning_rate': learning_rate})
  File "/home/alex/work/python/mxnet/python/mxnet/module/module.py", line 452, in init_optimizer
    assert self.binded and self.params_initialized
AssertionError
```

The main goal is to use this in a benchmark facility so any suggestions of how to do this properly in MXNet are welcome.

**Comment by salemmohammed** [comment]
@botev 
I am trying to print out loss value for each number of iteration. fit mother is confusing me. Can you tell how exactly it works ?

**Comment by yajiedesign** [comment]
This issue is closed due to lack of activity in the last 90 days. Feel free to reopen if this is still an active issue. Thanks!
